src/coreclr/vm/memberload.cpp

```diff
@@ -367,16 +367,23 @@ void MemberLoader::GetDescFromMemberRef(ModuleBase * pModule,
         {
             //
             // <NICE> this is duplicated logic GetFieldDescByIndex </NICE>
-            //
-            INDEBUG(mdFieldDef token = pFD->GetMemberDef();)
-
-            DWORD pos = static_cast<DWORD>(pFD - (pMT->GetApproxFieldDescListRaw() + pMT->GetNumIntroducedInstanceFields()));
-            _ASSERTE(pos >= 0 && pos < pMT->GetNumStaticFields());
+            //         
 
-            pFD = pMT->GetGenericsStaticFieldDescs() + pos;
-            _ASSERTE(pFD->GetMemberDef() == token);
-            _ASSERTE(!pFD->IsSharedByGenericInstantiations());
-            _ASSERTE(pFD->GetEnclosingMethodTable() == pMT);
+            DWORD pos = static_cast<DWORD>(pFD - (pMT->GetApproxFieldDescListRaw() + pMT->GetNumStaticFields()));
+            if(!(pos >= 0 && pos < pMT->GetNumStaticFields()))
+            {
+                DWORD fdIndex = pMT->GetIndexForFieldDesc(pFD);
```

Comment:
Are we assuming this is an EnC field? If so, we should assert that condition. It would help with documenting why the out of range index is okay and not an error.